### PR TITLE
Add Toggle Component

### PR DESCRIPTION
**Pull request by Krishna-13-cyber**
Signed-off-by: Krishna Narayanan <krishnanarayanan132002@gmail.com>

Depiction: https://asciinema.org/a/aIY1Aj86rx4asLjJVCXECNOCX

### CODE REVIEW

File: include/quick-ftxui.hpp [hunk 1]
```diff
@@ -18,7 +18,9 @@
 
 #include <iostream>
 #include <string>
+#include <vector>
 
+using namespace std;
```
Comment (SAtacker) [inline include/quick-ftxui.hpp]: Please remove this.

File: tests/test1.cpp [hunk 2]
```diff
@@ -19,23 +19,35 @@ auto parse_helper(std::string &&str) {
...
     REQUIRE(parse_helper("Horizontal{Slider{\"amool\", 2, 5, 100, 1}}"));
 
-    REQUIRE(parse_helper("Horizontal{           Button{          \"amool\"    ,       "
-                         "\"bmpp\" }           }"));
-    REQUIRE(parse_helper("Vertical{           Button{          \"amool\"    ,       "
-                         "\"bmpp\" }           }"));
+    REQUIRE(parse_helper("Vertical{Toggle{[" Opt1 " " Opt2 "], 1}}"));
+    REQUIRE(parse_helper("Horizontal{Toggle{[" Opt1 " " Opt2 "], 1}}"));
+    REQUIRE(parse_helper("Vertical{Toggle{[" Opt1 " " Opt2 "], 0}}"));
+    REQUIRE(parse_helper("Horizontal{Toggle{[" Opt1 " " Opt2 "], 0}}"));
```
Comment (SAtacker) [inline tests/test1.cpp:32]: Fix `" "` marks

File: tests/test1.cpp [hunk 3]
```diff
@@ -19,23 +19,35 @@ auto parse_helper(std::string &&str) {
...
     REQUIRE(!parse_helper("\"amool\"{Button{\"amool\",\"bmpp\"}}"));
     REQUIRE(!parse_helper("\"amool\"{_Button{\"amool\",\"bmpp\"}}"));
     REQUIRE(!parse_helper("Vertical{_Button{\"amool\",\"bmpp\"}_}"));
-    REQUIRE(!parse_helper("Vertical{Slider{\"amool\",\"bmpp\",\"cmqq\",\"dmrr\",\"emss\"}}"));
+    REQUIRE(!parse_helper(
+        "Vertical{Slider{\"amool\",\"bmpp\",\"cmqq\",\"dmrr\",\"emss\"}}"));
+    REQUIRE(parse_helper("Vertical{Toggle{[" Opt1 " " Opt2 " " Opt3 "], 1}}"));
+    REQUIRE(parse_helper("Vertical{Toggle{[" Opt1 " " Opt2 " " Opt3 " " Opt4
+                         "], 1}}"));
+    REQUIRE(parse_helper("Vertical{Toggle{[" Opt1 " " Opt2 "], 8}}"));
```
Comment (SAtacker) [inline tests/test1.cpp:50]: Fix `" "` marks

**Review by SAtacker — CHANGES_REQUESTED** [review]
Fix `" "` makrs

**Comment by Krishna-13-cyber** [comment]
Closing this PR as new issued #21

**Pull request closed without merging**